apps/indexer/src/providers/statusEndpointHandlers/grpc.ts

```diff
@@ -96,34 +91,24 @@ export async function fetchProviderStatusFromGRPC(provider: Provider, timeout: n
   };
 }
 
-async function queryStatus(hostUri: string, timeout: number): Promise<Status> {
+async function queryStatus(hostUri: string, timeoutMs: number): Promise<Status> {
   try {
-    return await createProviderClient(hostUri).getStatus(timeout);
+    return await getProviderSDK(hostUri).akash.provider.v1.getStatus({}, { timeoutMs });
   } catch (error) {
-    createProviderClient.cache.delete(hostUri);
+    getProviderSDK.cache.delete(hostUri);
     throw error;
   }
 }
 
-const createProviderClient = memoize((hostUri: string) => {
+const getProviderSDK = memoize((hostUri: string) => {
   // TODO: fetch port from chain
-  const url = hostUri.replace(":8443", ":8444").replace("https://", "dns:///");
-  const credentials = FakeInsecureCredentials.createInsecure();
-  const channel = new Channel(url, credentials, {
-    "grpc.keepalive_time_ms": minutesToMilliseconds(5),
-    "grpc.keepalive_permit_without_calls": 1
+  const url = hostUri.replace(":8443", ":8444");
+  return createProviderSDK({
+    // TODO: pass transport options
+    // "grpc.keepalive_time_ms": minutesToMilliseconds(5),
```

Comment:
question(non-blocking): is it better to just leave them there for now?

Reply:
There are 2 corresponding properties in Http2SessionManager which is used by new sdk. 

Analog to `grpc.keepalive_permit_without_calls`
<img width="839" height="213" alt="Image" src="https://github.com/user-attachments/assets/48b3aaf0-9ec9-4fba-bc58-b0722a5b857b" />

Analog to `grpc.keepalive_time_ms`
<img width="795" height="197" alt="Image" src="https://github.com/user-attachments/assets/df16b2f2-98b8-48f9-b83a-5106a1c1ee6d" />

Docs says that for stream request/response it sends ping requests automatically but for unary it doesn't. In this case we send unary requests to get status from provider. We fetch it every 10 secs, so having open connection I assume is useful to reduce latency

Reply:
I'll update it in the follow up PR

Reply:
follow up PR -> https://github.com/akash-network/console/pull/2103